### PR TITLE
Add FileSystemHelper tests

### PR DESCRIPTION
**Pull request by DanRStevens**
These tests demonstrate the problem I was seeing in the comment I left on Issue #78.

### CODE REVIEW

**Comment by DanRStevens** [comment]
We may want to merge in PR #80 to see the test results on CircleCI. We may also want to add the unit test to the Visual Studio test project file.

File: test/FileSystemHelper.test.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#include "FileSystemHelper.h"
+#include <gtest/gtest.h>
+
+#ifdef __cpp_lib_filesystem
+#include <filesystem>
+namespace fs = std::filesystem;
+#else
+#include <experimental/filesystem>
+namespace fs = std::experimental::filesystem;
+#endif
```
[the remark this answers is not in the document]
Comment (DanRStevens) [reply test/FileSystemHelper.test.cpp]: That's a very good point.

**Comment by DanRStevens** [comment]
Curiously the unit tests crash on CircleCI. When I do a native local run, or even a local run using the same Docker image and the `circleci` command line tool, the tests run but fail:
```
[ RUN      ] FileSystemHelper.GetGameDirectory
test/FileSystemHelper.test.cpp:8: Failure
Value of: fs::exists(gameDirectory)
  Actual: false
Expected: true
"Z:\\root\\project\\.build\\testBin\\"
test/FileSystemHelper.test.cpp:9: Failure
Value of: fs::is_directory(gameDirectory)
  Actual: false
Expected: true
"Z:\\root\\project\\.build\\testBin\\"
[  FAILED  ] FileSystemHelper.GetGameDirectory (49 ms)
```

**Comment by DanRStevens** [comment]
I saw some of the master builds fail with the same crash output. I don't think it's related to this branch. I suspect it's an intermittent failure, and the cause has already made it's way onto master.

With that said, I'm expecting some Linux failures with the unit tests, as that's what I'm seeing locally.

Will need to investigate the crash failures at some point. I checked a few addresses with OllyDbg. Some of the addresses on the stack backtrace didn't seem to make sense. Some seemed to suggest it may be related to CRT initialization of global variables.

**Comment by DanRStevens** [comment]
Work on this PR was previously blocked by random unit test crashes tracked by Issue #95 which was closed by PR #112.

The unit tests added here were to demonstrate a bug from Issue #78, which was closed by #87. The unit tests that demonstrated this bug may still be of value, despite the bug now being closed.

This PR is however now months out of date and contains merge conflicts. We may need to assess if this PR still contains a valuable contribution.

**Comment by Brett208** [comment]
I think these are generally valid tests that should be included. If they are not worth the time right now, perhaps we can just have them run on Windows only for now?

Thanks,
Brett

**Comment by DanRStevens** [comment]
I was tempted to do just that. I played around with it a bit, and it seemed like the filesystem calls just didn't work right on MinGW. Which would be a bit unexpected of course.

I played around a bit more, and have determined a pattern to the behavior. The `fs::exists` calls return false for paths with a trailing path separator. Unless of course it's just a drive letter, then it's the opposite, it returns false without the trailing path separator, but true when the trailing path separator is included.

Perhaps I can update the tests to work on all platforms now.

**Pull request merged**